### ISSUE TITLE
Add warmup_start_lr and eta_min to WarmupCosineSchedule

### ISSUE THREAD

**Issue by holgerroth** [Project-MONAI/MONAI#3989]
The current implementation of [WarmupCosineSchedule](https://github.com/Project-MONAI/MONAI/blob/fbd09a10cbdebff332aed4f95fd61dd28f0aee14/monai/optimizers/lr_scheduler.py#L59) will start the learning rate from 0. If step() is only called after the first epoch as [here](https://github.com/Project-MONAI/tutorials/blob/master/multimodal/openi_multilabel_classification_transchex/transchex_openi_multilabel_classification.ipynb), it causes the entire first epoch to be executed with a learning rate of zero.

Alternative implementations allow the user to set a `warmup_start_lr` and `eta_min` to avoid starting from or going down to zero, respectively, for example [here](https://lightning-bolts.readthedocs.io/en/stable/deprecated/schedulers/warmup_cosine_annealing.html#linear-warmup-cosine-annealing).

**Comment by wyli** [comment]
(reminds me that I had a similar comment a while ago for the upstream implementation... https://github.com/Project-MONAI/MONAI/issues/2183#issuecomment-842978322)

**Comment by ahatamiz** [comment]
The intended implementation requires `step()` to be called once before training initializes. Adding `eta_min` will alter the scheduler  behavior, hence is not desired. The proposed scheduler in my opinion better be added separately to avoid such issues.

**Comment by holgerroth** [comment]
You can still default eta_min to 0. Hence, the behavior of the scheduler would not change. Adding those arguments would just make a more flexible implementation available in monai.

**Comment by holgerroth** [comment]
I created a related issue for the tutorial https://github.com/Project-MONAI/tutorials/issues/630

**Comment by ahatamiz** [comment]
Hi @holgerroth,

Thanks for the comment. I respectfully disagree. The scheduler is generic and not designed for epoch-based or iteration-based training. In case of iteration-based, it works as desired. For epoch-based, user needs to call it once before training. Adding an extra parameter `eta_min` would deviate from original implementation and would not solve the epoch-based issue, if the default is set to be `0`. The user still needs to adjust `eta_min` , if the desired behavior is to start with a non-zero learning rate. 

I strongly recommend submitting a new PR for a scheduler specifically designed for epoch-based training. A good candidate maybe [here](https://github.com/PyTorchLightning/Lightning-Bolts/blob/master/pl_bolts/optimizers/lr_scheduler.py#L10-L120). 

As for the tutorial, I will update it to reflect this change to avoid zero learning rate in the first epoch.

**Comment by wyli** [comment]
addressed by https://github.com/Project-MONAI/MONAI/pull/5351
